src/tests/system/tests/test_sss_ssh_knownhosts.py

```diff
@@ -0,0 +1,122 @@
+"""
+sss_ssh_knownhosts tests.
+
+:requirement: IDM-SSSD-REQ: Status utility
+"""
+
+from __future__ import annotations
+
+import pytest
+from sssd_test_framework.roles.client import Client
+from sssd_test_framework.roles.ipa import IPA
+from sssd_test_framework.topology import KnownTopology
+
+
+@pytest.fixture(scope="module")
+def public_keys(moduledatadir: str) -> list[str]:
+    """
+    Read list of public keys from module data file.
+
+    :return: List of public keys.
+    :rtype: list[str]
+    """
+    keys: list[str] = []
+    with open(f"{moduledatadir}/public_keys") as f:
+        for line in f.readlines():
+            stripped = line.strip()
+            if stripped:
+                keys.append(stripped)
+
+    return keys
+
+
+@pytest.mark.ticket(gh=5518)
+@pytest.mark.importance("low")
+@pytest.mark.topology(KnownTopology.IPA)
+def test_sss_ssh_knownhosts__by_name(client: Client, ipa: IPA, public_keys: list[str]):
+    """
+    :title: sss_ssh_knownhosts returns public keys by name
+    :setup:
+        1. Create IPA host "ssh.ipa.test", public keys and IP resolvable via DNS
+        2. Enable ssh responder
+        3. Start SSSD
+    :steps:
+        1. Run "sss_ssh_knownhosts ssh.ipa.test"
+    :expectedresults:
+        1. All public keys were printed
+    :customerscenario: False
+    """
+    hostname = f"ssh.{ipa.domain}"
+    ip = "10.255.251.10"
+    ipa.host_account(hostname).add(ip=ip, sshpubkey=public_keys)
+
+    client.sssd.enable_responder("ssh")
+    client.sssd.start()
+
+    result = client.sss_ssh_knownhosts(hostname)
+    assert result.rc == 0
+    assert len(public_keys) == len(result.stdout_lines)
+    for key in public_keys:
+        assert f"{hostname},{hostname} {key}" in result.stdout_lines
```

Comment:
I think we should only check whether the hostname is present in the list of hostnames. Requiring `{hostname},{hostname}` might invalidate the test if we ever change that behavior.

Reply:
Yes, that's why I would like to produce a nice format with only "hostname key". Checking for hostname presence will not detect if we start returning completely invalid hostnames in the list.

Reply:
How about the following?
```
for pkey in public_keys:
    hosts, sep, key = pkey.partition(" ")
    assert hostname in hosts.split(",")
    assert key.strip() in keys
```

Reply:
Sure. But wouldn't it be better to patch the tool to produce only a single hostname now instead of complicating a test to accommodate for a change that will most likely never come? I mean it's literately two lines of code (I don't count braces).

Reply:
What you wrote will have to be much more complicated to be future proof. It does not check that there is no other hostname, it does not check that the output is indeed in the correct format so we can actually partition and split stuff.

Reply:
It's done.
@alexey-tikhonov because of this change, in `known_hosts()`, `tmp_ctx` became useless as it was used only once. So I did what you wanted and allocated the result directly on`mem_ctx` and completely removed the temporary context.

Reply:
And because I also forgot to remove the variable declaration, it compiled and even "worked"... Thanks.

Reply:
Thank you, this is even better. I changed and pushed the tests.